src/mc-fle2-payload-iev-private-v2.h

```diff
@@ -80,7 +80,39 @@
  *
  */
 
-typedef struct _mc_FLE2IndexedEncryptedValueV2_t mc_FLE2IndexedEncryptedValueV2_t;
+typedef enum {
+    kTypeInitV2,
```

Comment:
I think you're going to find that even these names are too generic given how C ODRs its enums into the global symbol space.  It wasn't a problem when these were scoped to a single TU, but now that they're in a header, we should give them more distinctness.

e.g.
```c
typedef enum {
  kFLE2IEVTypeInitV2,
  kFLE2IEVTypeEqualityV2,
  kFLE2IEVTypeRangeV2,
};
```

Reply:
Changed